org.opentosca.bus.management.invocation.plugin.remote/src/org/opentosca/bus/management/invocation/plugin/remote/ManagementBusInvocationPluginRemote.java

```diff
@@ -0,0 +1,230 @@
+
+                        final HashMap<String, String> outputParamMap = new HashMap<>();
+
+                        for (final KeyValueType outputParam : invocationResponse.getParams().getKeyValuePair()) {
+                            ManagementBusInvocationPluginRemote.LOG.debug("Key: {}, Value: {}", outputParam.getKey(),
+                                                                          outputParam.getValue());
+                            outputParamMap.put(outputParam.getKey(), outputParam.getValue());
+                        }
+                        message.setBody(outputParamMap, HashMap.class);
+                    } else {
+                        if (invocationResponse.getDoc() != null) {
+                            ManagementBusInvocationPluginRemote.LOG.debug("Response contains output as Document");
+
+                            try {
+                                final DocumentBuilderFactory dFact = DocumentBuilderFactory.newInstance();
+                                final DocumentBuilder build = dFact.newDocumentBuilder();
+                                final Document document = build.newDocument();
+
+                                final Element element = invocationResponse.getDoc().getAny();
+
+                                document.adoptNode(element);
+                                document.appendChild(element);
+
+                                message.setBody(document, Document.class);
+                            }
+                            catch (final Exception e) {
+                                ManagementBusInvocationPluginRemote.LOG.error("Unable to parse Document: {}",
+                                                                              e.getMessage());
+                            }
+                        } else {
+                            ManagementBusInvocationPluginRemote.LOG.warn("Response contains no output.");
+                            message.setBody(null);
+                        }
+                    }
+                } else {
+                    ManagementBusInvocationPluginRemote.LOG.error("Body contains no IAInvocationRequest object with the result.");
+                }
+            } else {
+                ManagementBusInvocationPluginRemote.LOG.error("Collaboration message contains no body.");
+            }
+        } else {
+            ManagementBusInvocationPluginRemote.LOG.error("Message has invalid class: {}",
+                                                          responseExchange.getIn().getBody().getClass());
+        }
+
+        return exchange;
+    }
+
+    @Override
+    public List<String> getSupportedTypes() {
+
+        // This plug-in supports only the special type 'remote' which is used to forward invocation
+        // requests to other OpenTOSCA Containers.
+        final List<String> types = new ArrayList<>();
+        types.add(Constants.REMOTE_TYPE);
+
+        return types;
```

Comment:
The whole block can be replaced with: `return Collections.singletonList(Constants.REMOTE_TYPE);`